openpype/plugins/actions/open_file_explorer.py

```diff
@@ -0,0 +1,172 @@
+        else:
+            self.open_in_explorer(path)
+
+    def get_workdir_data(self, project_doc, asset_doc, task_name, host_name):
+        """Mimic `openpype.lib.get_workdir_data` but require less data.
+
+        This allows for example to have only a valid `project_doc` passed in
+        and the rest being None. For all that is None a stub placeholder data
+        value will be returned.
+
+        Returns:
+            dict: Workdir data.
+
+        """
+
+        # Start with mostly stub placeholder data where we cannot match
+        # `openpype.lib.get_workdir_data` due to lack of input variables.
+        STUB = "<<NULL>>"
+        data = {
+            "project": {
+                "name": project_doc["name"],
+                "code": project_doc["data"].get("code")
+            },
+            "task": {
+                "name": STUB,
+                "type": STUB,
+                "short": STUB,
+            },
+            "asset": STUB,
+            "parent": STUB,
+            "hierarchy": STUB,
+            "app": STUB,
+            "user": getpass.getuser()
+        }
+
+        # Retrieve data similar to `openpype.lib.get_workdir_data` but only
+        # up to where we can. First using AVALON_ASSET in session.
+        if asset_doc:
+            asset_parents = asset_doc["data"]["parents"]
+            hierarchy = "/".join(asset_parents)
+
+            parent_name = project_doc["name"]
+            if asset_parents:
+                parent_name = asset_parents[-1]
+
+            # Insert asset data
+            data.update({
+                "asset": asset_doc["name"],
+                "parent": parent_name,
+                "hierarchy": hierarchy,
+            })
+
+        # Then insert task data when AVALON_TASK in session
+        if asset_doc and task_name is not None:
+            asset_tasks = asset_doc['data']['tasks']
+            task_type = asset_tasks.get(task_name, {}).get('type')
+            project_task_types = project_doc["config"]["tasks"]
+            task_code = project_task_types.get(task_type, {}).get(
+                "short_name"
+            )
+
+            # Insert task data
+            data["task"].update({
+                "name": task_name,
+                "type": task_type,
+                "code": task_code
+            })
+
+        return data
+
+    @staticmethod
+    def open_in_explorer(path):
+        platform_name = platform.system().lower()
+        if platform_name == "windows":
+            args = ["start", path]
+        elif platform_name == "darwin":
+            args = ["open", "-na", path]
+        elif platform_name == "linux":
+            args = ["xdg-open", path]
+        else:
+            raise RuntimeError(f"Unknown platform {platform.system()}")
+        # Make sure path is converted correctly for 'os.system'
+        os.system(subprocess.list2cmdline(args))
+
+    @staticmethod
+    def copy_path_to_clipboard(path):
+        from Qt import QtWidgets
```

Comment:
Likely needs to be `qtpy` nowadays.
```suggestion
        from qtpy import QtWidgets
```